engine/spls_qos.go

```diff
@@ -52,23 +53,47 @@ func (lcs *QOSSupplierSorter) SortSuppliers(prflID string, suppls []*Supplier,
 			return nil, err
 		}
 
-		srtData := map[string]interface{}{
+		srtData := map[string]float64{
 			utils.Weight: s.Weight,
 		}
-
-		for k, v := range metricSupp { //transfer data from metric into srtData
-			srtData[k] = v
+		for _, metric := range extraOpts.sortingParameters {
+			hasMetric := false                         //check if metricSupp have sortingParameter
+			for keyWithID, value := range metricSupp { //transfer data from metric into srtData
+				if metric == strings.Split(keyWithID, utils.InInFieldSep)[0] {
+					if val, hasKey := srtData[metric]; !hasKey ||
+						(metric == utils.MetaPDD && val < value) || //worst values
+						(metric != utils.MetaPDD && val > value) {
+						srtData[metric] = value
+						hasMetric = true
+					}
+				}
+			}
+			if hasMetric == false { //if not have populate with default value
+				srtData[metric] = -1
+				if metric == utils.MetaPDD {

```

Comment:
better do it with a case here since it could be we will have more in the future.